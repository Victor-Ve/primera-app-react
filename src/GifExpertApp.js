import React, { useState } from 'react';
import AddCategory from './components/AddCategory';
import GifGrid from './components/GifGrid';

const GifExpertApp = ({ defaultCategories = [] }) => {
  const [categorias, setCategorias] = useState(defaultCategories);

  return (
    <>
      <h2>Gif Expert app</h2>
      <AddCategory setCategorias={setCategorias} />
      <hr />
      <ol>
        {categorias.map((categoria) => (
          <GifGrid categoria={categoria} key={categoria} />
        ))}
      </ol>
    </>
  );
};

export default GifExpertApp;
