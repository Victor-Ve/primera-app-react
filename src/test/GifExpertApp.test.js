import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import GifExpertApp from '../GifExpertApp';

describe('Pruebas en <GifExpertApp />', () => {
  test('Macth con el snapshot', () => {
    const wrapper = shallow(<GifExpertApp />);

    expect(wrapper).toMatchSnapshot();
  });

  test('debe de mostrar una lista de categorias', () => {
    const categories = ['One puch', 'dragon ball'];
    const wrapper = shallow(
      <GifExpertApp defaultCategories={categories} />
    );

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('GifGrid').length).toBe(categories.length);
  });
});
