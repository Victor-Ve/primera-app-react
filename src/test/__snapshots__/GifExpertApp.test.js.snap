// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifExpertApp /> Macth con el snapshot 1`] = `
<Fragment>
  <h2>
    Gif Expert app
  </h2>
  <AddCategory
    setCategorias={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Pruebas en <GifExpertApp /> debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    Gif Expert app
  </h2>
  <AddCategory
    setCategorias={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      categoria="One puch"
      key="One puch"
    />
    <GifGrid
      categoria="dragon ball"
      key="dragon ball"
    />
  </ol>
</Fragment>
`;
